# <center>Análisis poblacional del panorama político y perspectivas para las elecciones presidenciales de 2022 en Colombia .</center>
$\color{black}{Juan\;Camilo\;García\;Malagón}$
## <center> $\color{blue}{\text{Business understanding}}$ </center>



$\color{blue}{\text{Introducción}}$

Hoy en día y desde siempre la política ha representado uno de los ambitos más marcados en la socidad, la postura política de un individuo usualmente recae en quien considera representa sus ideales (aunque esto no siempre ocurra);La política se encuentra intrínsecamente relacionada con poder y este resulta ser codiciado por un grupo puntual de individuos de la sociedad,especificamente los partidos politicos y a su vez de todos sus filiales y contribuyentes pues aunque la figura de Presidente represente un sustantivo singular de este dependen muchos de los altos cargos y toma de decisiones futuras para todos los habitantes.

Aunque la política en Colombia se ha distinguido por el bipartidismo desde la constitución del 1991 se da via libre al multipartidismo con lo cual los partidos políticos cobran una figura de gran influencia y a su vez los encargados de representarlos y competir con sus homólogos.En la actualidad compañias destacadas como GUARUMO,INVAMER,MOE y el CNC,Centro Nacional de Consultoría son los encargados de realizar encuestas ,sondeos y datos respecto al panorama político en Colombia proporcionando una visión general a la población respecto a las preferencias y favoritismos políticos de la mayoria ,pero además proporcionando la llave maestra a los partidos y sus canditatos respecto a cual grupo social de sufragantes podria ser crucial ganar en la siguiente contienda electoral.

El presente proyecto de Ciencia de datos empleando el proceso estándar entre industrias para la minería de datos (CRISP-DM) pretende estudiar las estadisticas políticas registradas en los ultimos periodos electorales dando un panorama de necesidades y preferencias de los ciudadanos el cual resulta de gran interés y utilidad para partidos politicos,agencias de publicidad,gremios sociales-económicos y otras diferentes esferas sociales.

$\color{blue}{\text{Importancia del problema}}$

$\color{blue}{\text{Estado del arte}}$

$Sistema\;Electoral\;Colombiano$

Sistema de eleccion popular donde los integrantes de los partidos politicos son escogidos para los diferentes cargos gubernamentales.

Se encuentra a cargo del CNE (Consejo Nacional Electoral)- y Registraduria Nacional del Estado Civil 
Velar por las normas electorales del Codigo electoral colombiano 

**CNE (Consejo Nacional Electoral)**

-Organismo autonomo e independiente parte de la organización electoral  establecido mediante el articulo 120 de la constitucion Politica Colombiana
-Se ecuentra conformado por 9 magistrados elegidos en Congreso

**Registraduria Nacional del Estado Civil**

Igualmente que el CNE es una institutcion de descentralizada de carácter publico encargada de la organización electoral 
Según establece el artículo 266 de la Constitución Política de Colombia, el Registrador Nacional del Estado Civil será escogido por los Presidentes de la Corte Constitucional, la Corte Suprema de Justicia y el Consejo de Estado, mediante concurso de méritos organizado según la ley

**Codigo electoral colombiano**

Actualmente rige el codigo electoral establecido en el decredto 2241 de 1986 en el cual se estipulan autoridades electorales, delegados, partes,funciones,disposiciones y capacidades de las distintas partes de los miembros electorales 
Actualmente se encuentra en tramite como ley estatutaria un codigo electoral en Congreso (Camara y Senado deciden)



$\color{blue}{\text{Business Objectives and Success Criteria}}$

**Objetivo general:**

Analizar los comportamientos electorales de la elección de la figura Presidencial en Colombia en los ultimos años mediante los datos registrados de sufragios en las anteriores contiendas analizando variables como la dominancia politica,la representatividad y no representatividad,voter turnout,ergonomía electoral entre otras variables contrastando con las proyecciones actuales de los candidatos presidenciales para el periodo electoral 2022-2026


**Objetivos especificos:**

- Dar un panorama político por departamentos con base en sus comportamientos electorales previos
- Adquirir bases de datos referentes a las elecciones de los periodos 2014-2018 y 2018-2022 de los datos públicos  recopilados por gobierno y registraduría
- Investigar los diferentes patrones de distribuciones de probabilidad asociados a las encuestas y resultados hechos durante las anteriores campañas electorales


**Criterios de éxito del Proyecto**

- Tener un rango de proximidad con base en el análisis hecho frente a las encuestas actuales para los distintos departamentos y regiones
- Concluir en tiempo oportuno el análisis electoral propuesto
- Obtener posibles escenarios de los cuales se puedan derivar estrategias de utilidad para los distintos actores interesados en la próxima contienda electoral
- Lograr ajustes en los patrones de distribuciones de probabilidad de buena calidad con los datos recopilados



$Resource\;Inventory$

- Computadora con lenguaje de programación Python 3.8.5 y librerias necesarias.
- Datos proporcionados por entidades gubernamentales sobre resultados y sondeos de elecciones, preferencias partidarias y candidaduras anteriores.
- Equipo de expertos y analistas en contiendas políticas y asesores de campañas.

$Requirements, Assumptions, and\;Constraints$

**Requirements**
- Los datos son de caracter público para una mayor reproducibilidad.
- En principio no existe obligación o requerimiento contractual de publicar o compartir los datos obtenidos de este proyecto sin embargo existe la posibilidad de implementarlo en un tablero de control.

**Assumptions**
- El proyecto responde a una actividad netamente académica sin ánimo de lucro por lo cual no correlaciona ningún factor económico
- Los datos pueden presentar un sesgo estadistico debido a los margenes de error y posibles errores humanos en los conteos.
- Para los posibles clientes o patrocinadores son mas fructiferos los resultados derivados de este estudio como posibles estrategias de campaña más que el proyecto en si mismo

**Constraints**
- Los únicos requerimientos legales del tratamiento de las bases de datos usadas es no comprotemer de forma directa o directa alguna entidad gubernamental o en nombre de esta lo cual claramente no se pretende al ser un ejercicio académico

$Risks\;and\;Contingencies$

- Programación: Ante un posible retraso del proyecto se considera la reprogramación del calendario previsto para la mayor realización porcentual del proyecto

- Financiero: No aplica

- Datos: Existe un margen de error respecto a la calidad de los datos proporcionados pues algunos de estos por su magnitud aún estan sujetos a verifación

- Resultados: Existe la posibilidad de escenarios mas leves o con menos información que los planteados inicialmente los cuales no dejan de contener información valiosa para los potenciales clientes a primera estancia.


$Cost\;/Benefit\;Analysis$

**Costos**

De forma ideal se plantea la necesidad de:
- Un consultor conocedor del panorama politico
- Un cientifico de datos para el tratamiento y manejo de la BigData
- Un equipo de encuestadores para el muestro y verifación de posibles escenarios en determinados grupos poblacionales

**Beneficios**

- Poder generar un planteamiento óptimo de recursos de campaña 
- A partir de los resultados plantear ideas ,estrategias y propuestas optimas para ganar sufragantes en las poblaciones espeficas.

$Determining\;Data\;Mining\;Goals$

- Recopilar los datos históricos de periodos anteriores para realizar el data screening de los mismos (Market basket analysis):
- Establecer ajustes a funciones de densidad de probabilidad para las diferentes variables asociadas a las bases de datos de encuestas y resultados de campañas electorales anteriores (Profiling)
- Proyectar escenarios electorales bien ya sea un posible victoria de un candidato en primera vuelta o una posible contienda en segunda vuelta de dos de ellos (Sequence analysis)

**Criterios de éxito del Proyecto**

- Obtener posibles escenarios de los cuales se puedan derivar estrategias de utilidad para los distintos actores interesados en la próxima contienda electoral
- Lograr ajustes en los patrones de distribuciones de probabilidad con una buena bondad de ajuste medida a través de diferentes test 
- Establecer una métrica que permita cuantificar la certeza de los resultados con unos umbrales de desviación.


$Project\;Plan$


| Fase | Tiempo | Recursos | Riesgos |
|:-:|:-:|:-:|:-:|
| Entendimiento del negocio | 1 semana | Todos los análisis | Cambio de presupuesto para  la ejecución del proyecto |
| Entendimiento de los datos | 2 semana | Equipos de muestreo y sondeo de  intenciones de voto y consultor  político | Problema en la recopilación de datos, problemas técnicos y sesgos estadísticos |
| Preparación de los datos | 2 semanas | Científico y analista de datos | Problema en los datos,  problemas técnicos |
| Evaluación | 2 semana | Consultor político y científico de   datos | Cambios de presupuesto,  inhabilidad de publicación  de proyecciones del modelo |

## <center> $\color{blue}{\text{Data understanding}}$ </center>


**Data Quantity**

* *Formato*: Data estructurada en archivo xlsx
* *Fuente* : Registraduría Nacional del Estado Civil
* *Obtención* : Escrutinio de mesas de votantes
    * Primera Vuelta: https://www.registraduria.gov.co/IMG/xlsx/PRE_1ERA_VUELTA_MMV_2014.xlsx
        * 704176 Filas x 16 Columnas

    * Segunda Vuelta: https://www.registraduria.gov.co/IMG/xlsx/PRE_2DA_VUELTA_MMV_2014.xlsx

        * 456253 Filas x 14 Columnas

In [1]:
import pandas as pd
import numpy as np

In [2]:
URL='DATOS_01_14.xlsx'
Datos_01_2014=pd.read_excel(URL)

In [3]:
Datos_01_2014.columns

Index(['DIVIPOL_MESA', 'DIVIPOL', 'COD_DEP', 'DEPARTAMENTO', 'COD_MUN',
       'MUNICIPIO', 'ZONA', 'PUESTO', 'MESA', 'JAL', 'BOLETIN', 'COD_PARTIDO',
       'NOMB_PARTIDO', 'COD_CANDIDATO', 'NOMB_CANDIDATO', 'VOTOS'],
      dtype='object')

In [4]:
Datos_01_2014

DIVIPOL_MESA  DIVIPOL  COD_DEP DEPARTAMENTO  COD_MUN  MUNICIPIO  ZONA  \
0        10010101-1     1001        1    ANTIOQUIA        1   MEDELLIN     1   
1        10010101-1     1001        1    ANTIOQUIA        1   MEDELLIN     1   
2        10010101-1     1001        1    ANTIOQUIA        1   MEDELLIN     1   
3        10010101-1     1001        1    ANTIOQUIA        1   MEDELLIN     1   
4        10010101-1     1001        1    ANTIOQUIA        1   MEDELLIN     1   
...             ...      ...      ...          ...      ...        ...   ...   
704171  888157586-1    88815       88   CONSULADOS      815  VENEZUELA    75   
704172  888157586-1    88815       88   CONSULADOS      815  VENEZUELA    75   
704173  888157586-1    88815       88   CONSULADOS      815  VENEZUELA    75   
704174  888157586-1    88815       88   CONSULADOS      815  VENEZUELA    75   
704175  888157586-1    88815       88   CONSULADOS      815  VENEZUELA    75   

       PUESTO  MESA  JAL  BOLETIN  COD_PARTIDO  \
0           1     1    0     9999            1   
1           1     1    0     9999            2   
2           1     1    0     9999            3   
3           1     1    0     9999            4   
4           1     1    0     9999            5   
...       ...   ...  ...      ...          ...   
704171     86     1    0     9999            4   
704172     86     1    0     9999            5   
704173     86     1    0     9999            0   
704174     86     1    0     9999            0   
704175     86     1    0     9999            0   

                                             NOMB_PARTIDO  COD_CANDIDATO  \
0       POLO DEMOCRÁTICO ALTERNATIVO PDA UNIÓN PATRÍOT...              1   
1                          PARTIDO CONSERVADOR COLOMBIANO              2   
2                                         UNIDAD NACIONAL              3   
3                                   PARTIDO ALIANZA VERDE              4   
4            CENTRO DEMOCRÁTICO MANO FIRME CORAZÓN GRANDE              5   
...                                                   ...            ...   
704171                              PARTIDO ALIANZA VERDE              4   
704172       CENTRO DEMOCRÁTICO MANO FIRME CORAZÓN GRANDE              5   
704173                                                NaN            996   
704174                                                NaN            997   
704175                                                NaN            998   

                     NOMB_CANDIDATO  VOTOS  
0                       CLARA LÓPEZ      9  
1              MARTHA LUCÍA RAMÍREZ     15  
2       JUAN MANUEL SANTOS CALDERÓN     37  
3                  ENRIQUE PEÑALOSA      2  
4                OSCAR IVÁN ZULUAGA     42  
...                             ...    ...  
704171             ENRIQUE PEÑALOSA      8  
704172           OSCAR IVÁN ZULUAGA    105  
704173              VOTOS EN BLANCO      1  
704174                  VOTOS NULOS      1  
704175            VOTOS NO MARCADOS      0  

[704176 rows x 16 columns]

In [5]:
URL='DATOS_02_14.xlsx'
Datos_02_2014=pd.read_excel(URL)

In [6]:
Datos_02_2014.columns

Index(['COD_DEP', 'DEPARTAMENTO', 'COD_MUN', 'MUNICIPIO', 'ZONA', 'PUESTO',
       'MESA', 'JAL', 'BOLETIN', 'COD_PARTIDO', 'NOMBRE_PARTIDO',
       'COD_CANDIDATO', 'NOMBRE_CANDIDATO', 'VOTOS'],
      dtype='object')

In [7]:
Datos_02_2014

COD_DEP DEPARTAMENTO  COD_MUN  MUNICIPIO  ZONA PUESTO  MESA  JAL  \
0             1    ANTIOQUIA        1   MEDELLIN     1      1     1    0   
1             1    ANTIOQUIA        1   MEDELLIN     1      1     1    0   
2             1    ANTIOQUIA        1   MEDELLIN     1      1     1    0   
3             1    ANTIOQUIA        1   MEDELLIN     1      1     1    0   
4             1    ANTIOQUIA        1   MEDELLIN     1      1     1    0   
...         ...          ...      ...        ...   ...    ...   ...  ...   
446252       88   CONSULADOS      815  VENEZUELA    75     86     1    0   
446253       88   CONSULADOS      815  VENEZUELA    75     86     1    0   
446254       88   CONSULADOS      815  VENEZUELA    75     86     1    0   
446255       88   CONSULADOS      815  VENEZUELA    75     86     1    0   
446256       88   CONSULADOS      815  VENEZUELA    75     86     1    0   

        BOLETIN  COD_PARTIDO                                NOMBRE_PARTIDO  \
0          9999            1                               UNIDAD NACIONAL   
1          9999            2  CENTRO DEMOCRÁTICO MANO FIRME CORAZÓN GRANDE   
2          9999            0                                           NaN   
3          9999            0                                           NaN   
4          9999            0                                           NaN   
...         ...          ...                                           ...   
446252     9999            1                               UNIDAD NACIONAL   
446253     9999            2  CENTRO DEMOCRÁTICO MANO FIRME CORAZÓN GRANDE   
446254     9999            0                                           NaN   
446255     9999            0                                           NaN   
446256     9999            0                                           NaN   

        COD_CANDIDATO             NOMBRE_CANDIDATO  VOTOS  
0                   1  JUAN MANUEL SANTOS CALDERÓN     64  
1                   2           ÓSCAR IVÁN ZULUAGA     79  
2                 996              VOTOS EN BLANCO      4  
3                 997                  VOTOS NULOS      3  
4                 998            VOTOS NO MARCADOS      3  
...               ...                          ...    ...  
446252              1  JUAN MANUEL SANTOS CALDERÓN    181  
446253              2           ÓSCAR IVÁN ZULUAGA    164  
446254            996              VOTOS EN BLANCO      1  
446255            997                  VOTOS NULOS      1  
446256            998            VOTOS NO MARCADOS      1  

[446257 rows x 14 columns]

**Data Quality**

Los datos proporcionan la cantidad de votos registrados por ubicación geográfica y por partido político siendo de gran utilidad para el presente análisis pues se concentran el número de sufragantes afines a un partido político por región

* Tipo :Se presentan datos en su mayoría númericos y cadenas de caracteres

**Notas**

* Es importante tener en cuenta el número de filas del archivo puesto que la lectura de estas requiere un tiempo considerable de procesamiento en Jupyter
* Se toman en cuenta las bases de codigo abierto publicadas del año 2014 pues hasta la fecha no se tiene una data estructurada de los formatos E-24 Y E-26 del año 2018,periodo anterior.

In [8]:
import pandas as pd
import openpyxl
wb = openpyxl.load_workbook('METADATOS.xlsx')
ws = wb['meta_data']
md = pd.DataFrame(ws.values)
md.columns = md.iloc[0]
md = md.drop(0)
md.head(24)

0         Metadato                                        Descripción  \
1     DIVIPOL_MESA  Corresponde al codigo DIVIPOL adjuntando el nú...   
2          DIVIPOL   Es la división político administrativa en la ...   
3          COD_DEP  Corresponde a un número asignado por el DANE a...   
4     DEPARTAMENTO                            Nombre del departamento   
5          COD_MUN  Corresponde al un número asignado por el DANE ...   
6        MUNICIPIO                               Nombre del municipio   
7             ZONA  Corresponde al número de división establecida ...   
8           PUESTO  Corresponde la número de puesto de votación de...   
9             MESA  Corresponde al número de mesa dispuesto en cad...   
10     COD_PARTIDO  Corresponde al número asignado a los distintos...   
11    NOMB_PARTIDO  Nombre del partido politico al cual pertenece ...   
12   COD_CANDIDATO  Código asignado al candidado de cada partido e...   
13  NOMB_CANDIDATO  Nombre del candidado seleccionado por el parti...   
14           VOTOS  Número total de votos registrados por cada opc...   

0  Unida de medición  
1    Texto ( String)  
2    Número(Integer)  
3    Número(Integer)  
4    Texto ( String)  
5    Número(Integer)  
6    Texto ( String)  
7    Número(Integer)  
8    Número(Integer)  
9    Número(Integer)  
10   Número(Integer)  
11   Texto ( String)  
12   Número(Integer)  
13   Número(Integer)  
14   Número(Integer)

<img src="diagramaflujo.png">

**Limpieza y filtración de Datos**

* Primeramente de los datos originales se obseva la no variación en ninguna celda de las columnas 'JAL' e 'Informe' pues los datos se encuentran clasificados por código-zona por lo cual el dato de la Junta Local Administrativa a la cual se encuentra adscrita la zona no es relevante y además estos datos corresponden al conteo voto a voto final por la registraduría por lo cual corresponden todos solo a un mismo informe,el último y definitivo.

* Es importante destacar que los datos 'voto en blanco','voto nulo' y 'voto no marcado' claramente no corresponden a ninguna partido político por lo cual las respectivas celdas se encuentran vacias


## $\color{red}{--------------------------------------}$
## $\color{red}{Observaciones}$

- Se realizó una nueva busqueda de datos pues los suministrados por la registraduría y datos abiertos del gobierno presentan un sesgo temporal considerable y no presentan información representativa de una forma sólida .
- Como lo planteado el Business Understanding se plantea la toma y procesamiento de datos de entidades externas como el caso de la Misión de Observación electoral la cual posee mayor variedad de datos y de mayor calidad por lo cual para obtener un mayor nivel de éxito en el presente proyecto de análisis de datos se considera necesario actualizar los módulos en lo que a data representan como los son Data Understading,MetaData,DataCleaning y DataScreening

## $\color{red}{--------------------------------------}$

## <center> $\color{blue}{\text{Data understanding}}$ </center>

In [9]:
import pandas as pd
import openpyxl
wb = openpyxl.load_workbook('DataSet.xlsx')
ws = wb['Hoja1']
md = pd.DataFrame(ws.values)
md.columns = md.iloc[0]
md = md.drop(0)
md.head(24)

0    CARACTERISTICA   NaN                                            DATASET  \
1           ARCHIVO  None                        2014PresidentePrimeraVuelta   
2         CONTENIDO  None  Votación Presidencial 2014 Primera vuelta por ...   
3       DESCRIPCIÓN  None  Tabla de datos registrados en las elecciones p...   
4      CONSTRUCCIÓN  None                Misión de Observación Electoral MOE   
5   FUENTE DE DATOS  None            Registraduría Nacional del Estado Civil   
6          KEYWORDS  None       Departamentos,Paticipación, Votos,Candidatos   
7      LOCALIZACIÓN  None  https://www.datoselectorales.org/wp-content/up...   
8             FECHA  None                                  Noviembre de 2014   
9           FORMATO  None                                               XLSX   
10         DERECHOS  None                                     Datos Abiertos   
11             None  None                                               None   

0                                                 NaN  \
1                         2014PresidenteSegundaVuelta   
2   Votación Presidencial 2014 Primera vuelta por ...   
3   Tabla de datos registrados en las elecciones p...   
4                 Misión de Observación Electoral MOE   
5             Registraduría Nacional del Estado Civil   
6        Departamentos,Paticipación, Votos,Candidatos   
7   https://www.datoselectorales.org/wp-content/up...   
8                                   Noviembre de 2014   
9                                                XLSX   
10                                     Datos Abiertos   
11                                               None   

0                                                 NaN  \
1                         2018PresidentePrimeraVuelta   
2   Votación Presidencial 2018 Primera vuelta por ...   
3   Tabla de datos registrados en las elecciones p...   
4                 Misión de Observación Electoral MOE   
5             Registraduría Nacional del Estado Civil   
6        Departamentos,Paticipación, Votos,Candidatos   
7   https://www.datoselectorales.org/wp-content/up...   
8                                       Enero de 2019   
9                                                XLSX   
10                                     Datos Abiertos   
11                                               None   

0                                                 NaN  \
1                         2018PresidenteSegundoVuelta   
2   Votación Presidencial 2018 Primera vuelta por ...   
3   Tabla de datos registrados en las elecciones p...   
4                 Misión de Observación Electoral MOE   
5             Registraduría Nacional del Estado Civil   
6        Departamentos,Paticipación, Votos,Candidatos   
7   https://www.datoselectorales.org/wp-content/up...   
8                                       Enero de 2019   
9                                                XLSX   
10                                     Datos Abiertos   
11                                               None   

0                                                 NaN  \
1                               LibroResultadosPres14   
2             Resultados Consolidados Elecciones 2014   
3   Análisis poblacional de participación electora...   
4   Misión de Observación Electoral MOE Agencia de...   
5             Registraduría Nacional del Estado Civil   
6   Partidos,Candidatos,Contiendas, Participación,...   
7                                 https://moe.org.co/   
8                                     Febrero de 2015   
9                                                 PDF   
10                                     Datos Abiertos   
11                                               None   

0                                                 NaN  
1                               LibroResultadosPres18  
2             Resultados Consolidados Elecciones 2018  
3   Análisis poblacional de participación electora...  
4   Misión de Observación Electoral MOE Agencia de...  
5             Regis

In [10]:
URL='2014PresidentePrimeraVuelta.xlsx'
HOJA='Presidencia 1raV 2014'
Datos_10_2014=pd.read_excel(URL,sheet_name=HOJA)

In [11]:
Datos_10_2014

Code DANE       Municipio        Depto  Censo electoral  \
0            5001        Medellín    Antioquia          1412495   
1            5002       Abejorral    Antioquia            14939   
2            5004        Abriaquí    Antioquia             1903   
3            5021      Alejandría    Antioquia             4377   
4            5030           Amagá    Antioquia            20893   
...           ...             ...          ...              ...   
1118        99001  Puerto Carreño      Vichada            15621   
1119        99524    La Primavera      Vichada             7190   
1120        99624   Santa Rosalía      Vichada             2893   
1121        99773        Cumaribo      Vichada            18562   
1122  Consulados       Consulados  Consulados            851066   

      Participación  %Participación  Abstención   %Abstención  Votos válidos  \
0            700180        0.495704       712315     0.504296         682458   
1              4837        0.323783        10102     0.676217           4694   
2               709        0.372570         1194     0.627430            681   
3              1720        0.392963         2657     0.607037           1675   
4              7430        0.355622        13463     0.644378           7111   
...             ...             ...          ...          ...            ...   
1118           4880        0.312400        10741     0.687600           4805   
1119           2880        0.400556         4310     0.599444           2848   
1120           1118        0.386450         1775     0.613550           1105   
1121           3156        0.170025        15406     0.829975           3076   
1122         101002        0.118677       750064     0.881323         100310   

      %Votos váldos  ... Votos Enrique Peñalosa  %Votos Enrique Peñalosa  \
0          0.974689  ...                  43931                 0.064372   
1          0.970436  ...                     86                 0.018321   
2          0.960508  ...                     12                 0.017621   
3          0.973837  ...                     32                 0.019104   
4          0.957066  ...                    315                 0.044298   
...             ...  ...                    ...                      ...   
1118       0.984631  ...                    218                 0.045369   
1119       0.988889  ...                     55                 0.019312   
1120       0.988372  ...                     20                 0.018100   
1121       0.974651  ...                     95                 0.030884   
1122       0.993149  ...                  14015                 0.139717   

      Votos en Blanco  %Votos en Blanco  Votos inválidos  %Votos inválidos  \
0               55407          0.081187            17722          0.025311   
1                 184          0.039199              143          0.029564   
2                  20          0.029369               28          0.039492   
3                 154          0.091940               45          0.026163   
4                 523          0.073548              319          0.042934   
...               ...               ...              ...               ...   
1118              160          0.033299               75          0.015369   
1119               69          0.024228               32          0.011111   
1120               21          0.019005               13          0.011628   
1121               79          0.025683               80          0.025349   
1122             4444          0.044303              692          0.006851   

      Votos Nulos  %Votos Nulos  Tarjetones no marcados  \
0           16264      0.023228                    1458   
1              93      0.019227                      50   
2              12      0.016925                      16   
3              15      0.008721                      30   
4             269      0.036205                      50   
...           ...           ...       

In [12]:
URL='2014PresidenteSegundaVuelta.xlsx'
HOJA='Presidencia 2daV 2014'
Datos_20_2014=pd.read_excel(URL,sheet_name=HOJA)

In [13]:
Datos_20_2014

Code DANE       Municipio Departamento  Censo electoral  \
0          5001        Medellín    Antioquia          1412495   
1          5002       Abejorral    Antioquia            14939   
2          5004        Abriaquí    Antioquia             1903   
3          5021      Alejandría    Antioquia             4377   
4          5030           Amagá    Antioquia            20893   
...         ...             ...          ...              ...   
1117      97889       Yavaraté        Vaupés              994   
1118      99001  Puerto Carreño      Vichada            15621   
1119      99524    La Primavera      Vichada             7190   
1120      99624   Santa Rosalía      Vichada             2893   
1121      99773        Cumaribo      Vichada            18562   

      Participación / Total Votación  Abstención  \
0                             738993      673502   
1                               6933        8006   
2                                875        1028   
3                               2206        2171   
4                               9241       11652   
...                              ...         ...   
1117                             333         661   
1118                            5753        9868   
1119                            3264        3926   
1120                            1279        1614   
1121                            3880       14682   

      Juan Manuel Juan Manuel Santos  Oscar Iván Oscar Iván Zuluaga  \
0                             208597                         447208   
1                               1262                           5300   
2                                308                            514   
3                               1120                            959   
4                               3494                           4696   
...                              ...                            ...   
1117                             247                             83   
1118                            2648                           2808   
1119                             745                           2403   
1120                             416                            848   
1121                            2744                           1022   

      Votos en Blanco  Votos Válidos  ...  %Participación  %Abstención  \
0               53554         709359  ...           52.32        47.68   
1                 186           6748  ...           46.41        53.59   
2                  30            852  ...           45.98        54.02   
3                  84           2163  ...           50.40        49.60   
4                 624           8814  ...           44.23        55.77   
...               ...            ...  ...             ...          ...   
1117                1            331  ...           33.50        66.50   
1118              197           5653  ...           36.83        63.17   
1119               79           3227  ...           45.40        54.60   
1120                8           1272  ...           44.21        55.79   
1121               50           3816  ...           20.90        79.10   

      %Juan Manuel Juan Manuel Santos  %Oscar Iván Oscar Iván Zuluaga  \
0                               29.41                           63.04   
1                               18.70                           78.54   
2                               36.15                           60.33   
3                               51.78                           44.34   
4                               39.64                           53.28   
...                               ...                             ...   
1117                            74.62                           25.08   
1118                            46.84                           49.67   
1119                            23.09                           74.47   
1120                            32.70                           66.67   
1121                            71.91                  

In [14]:
URL='2018PresidentePrimeraVuelta.xlsx'
HOJA='Presidencia 1raV 2018'
Datos_10_2018=pd.read_excel(URL,sheet_name=HOJA)

In [15]:
Datos_10_2018

Code       Municipio       Depto  Censo electoral   \
0       5001        Medellín   Antioquia           1588150   
1       5002       Abejorral   Antioquia             14967   
2       5004        Abriaquí   Antioquia              1913   
3       5021      Alejandría   Antioquia              4549   
4       5030           Amagá   Antioquia             22350   
...      ...             ...         ...               ...   
1118   99001  Puerto Carreño     Vichada             18553   
1119   99524    La Primavera     Vichada              7116   
1120   99624   Santa Rosalía     Vichada              3197   
1121   99773        Cumaribo     Vichada             19585   
1122  999999      Consulados  Cansulados            812071   

      Participación/Total votación   Abstención        Ganador  Votos Válidos  \
0                           1009878      578272     Iván Duque         998455   
1                              6582        8385     Iván Duque           6408   
2                               994         919     Iván Duque            985   
3                              2203        2346     Iván Duque           2174   
4                             10460       11890     Iván Duque          10267   
...                             ...         ...            ...            ...   
1118                           7020       11533     Iván Duque           6941   
1119                           3277        3839     Iván Duque           3243   
1120                           1498        1699     Iván Duque           1491   
1121                           4188       15397  Gustavo Petro           4105   
1122                         280102      531969     Iván Duque         279089   

      Iván Duque  Gustavo Petro  ...  %Germán Vargas Lleras  \
0       535239.0          77067  ...               0.024170   
1         4580.0            184  ...               0.054151   
2          828.0             11  ...               0.027411   
3         1513.0             72  ...               0.057958   
4         5470.0            598  ...               0.043830   
...          ...            ...  ...                    ...   
1118      4018.0            829  ...               0.125198   
1119      2393.0            261  ...               0.105766   
1120      1008.0            107  ...               0.156271   
1121      1037.0           2046  ...               0.136419   
1122    152276.0          34446  ...               0.037404   

      %Humberto De La Calle  %Jorge Antonio Trujillo   %Viviane Morales  \
0                  0.025936                  0.002200          0.001298   
1                  0.011392                  0.002341          0.003745   
2                  0.015228                  0.002030          0.002030   
3                  0.043238                  0.000460          0.003680   
4                  0.017240                  0.001461          0.001266   
...                     ...                       ...               ...   
1118               0.016424                  0.004466          0.003170   
1119               0.004934                  0.004009          0.000308   
1120               0.010060                  0.000000          0.000671   
1121               0.014860                  0.006090          0.005116   
1122               0.015371                  0.001419          0.001455   

      %Promotores Voto en Blanco  %Voto en Blanco   %Total Voto en Blanco  \
0                       0.001850          0.020381               0.022231   
1                       0.001717          0.012016               0.013733   
2                       0.001015          0.010152               0.011168   
3                       0.001380          0.020699               0.022079   
4                       0.002045          0.021915               0.023960   
...                          ...               ...                    ...   
1118                    0.000720          0.013687               0.014407   
1119            

In [16]:
URL='2018PresidenteSegundaVuelta.xlsx'
HOJA='Presidente 2daV 2018'
Datos_20_2018=pd.read_excel(URL,sheet_name=HOJA)

In [17]:
Datos_20_2018

Code       Municipio        Depto  Censo electoral   \
0       5001        Medellín    Antioquia           1588150   
1       5002       Abejorral    Antioquia             14967   
2       5004        Abriaquí    Antioquia              1913   
3       5021      Alejandría    Antioquia              4549   
4       5030           Amagá    Antioquia             22350   
...      ...             ...          ...               ...   
1118   99001  Puerto Carreño      Vichada             18553   
1119   99524    La Primavera      Vichada              7116   
1120   99624   Santa Rosalía      Vichada              3197   
1121   99773        Cumaribo      Vichada             19585   
1122  999999     Consulados   Consulados             812071   

      Participación/Total votación   Abstención        Ganador  Votos Válidos  \
0                            979252      608898     Ivan Duque         965392   
1                              7397        7570     Ivan Duque           7293   
2                              1056         857     Ivan Duque           1045   
3                              2343        2206     Ivan Duque           2318   
4                             10755       11595     Ivan Duque          10507   
...                             ...         ...            ...            ...   
1118                           6608       11945     Ivan Duque           6520   
1119                           3229        3887     Ivan Duque           3208   
1120                           1462        1735     Ivan Duque           1459   
1121                           4846       14739  Gustavo Petro           4782   
1122                         259497      552574     Ivan Duque         258848   

      Iván Duque  Gustavo Petro  ...  Votos inválidos  \
0         698505         207642  ...            13860   
1           6572            547  ...              104   
2            975             43  ...               11   
3           2018            216  ...               25   
4           8192           1706  ...              248   
...          ...            ...  ...              ...   
1118        4721           1580  ...               88   
1119        2768            361  ...               21   
1120        1216            210  ...                3   
1121        1469           3213  ...               64   
1122      180948          69553  ...              649   

      %Participación electoral   %Abstención  %Votos Válidos  %Iván Duque  \
0                      0.616599     0.383401        0.985846     0.723545   
1                      0.494221     0.505779        0.985940     0.901138   
2                      0.552013     0.447987        0.989583     0.933014   
3                      0.515058     0.484942        0.989330     0.870578   
4                      0.481208     0.518792        0.976941     0.779671   
...                         ...          ...             ...          ...   
1118                   0.356169     0.643831        0.986683     0.724080   
1119                   0.453766     0.546234        0.993496     0.862843   
1120                   0.457304     0.542696        0.997948     0.833448   
1121                   0.247434     0.752566        0.986793     0.307194   
1122                   0.319550     0.680450        0.997499     0.699051   

      %Gustavo Petro  %Voto en Blanco   %Nulos   %Tarjetones No Marcados  \
0           0.215086         0.061369  0.001074                 0.013079   
1           0.075003         0.023858  0.004867                 0.009193   
2           0.041148         0.025837  0.009470                 0.000947   
3           0.093184         0.036238  0.004695                 0.005975   
4           0.162368         0.057961  0.002417                 0.020642   
...              ...              ...       ...                      ...   
1118        0.242331         0.033589  0.001967                 0.011350   
1119        0.112531         0.024626  0.001548             

## <center> $\color{blue}{\text{Data analysis}}$ </center>

Para entender los datos obtenidos primero debemos entender la definción de las variables presentadas en los datos, así como su jerarquia y correlaciones para un mejor entendimientos de estos y sus implicaciones.

* **Code:**

La Codificación de la División Político Administrativa de Colombia es una nomenclatura estandarizada, diseñada por el DANE para la identificación de entidades territoriales (departamentos, distritos y municipios),áreas no municipalizadas y centros poblados, mediante la asignación de un código numérico único.

* **Municipio**

Los municipios de Colombia corresponden al segundo nivel de división administrativa en Colombia, que mediante agrupación conforman los departamentos.Colombia posee 1103 municipios registrados en el DANE (conteo que incluye los 8 distritos especiales que se cuentan también como municipios), más las 18 áreas no municipalizadas y la isla de San Andrés, en total son 1122 entidades administrativas locales.En Colombia los recursos pasan de la nación al departamento y de este al municipio. Cabe resaltar que el Distrito Capital funciona al nivel de un Departamento.

* **Departamento**

La Constitución de 1991 establece a Colombia como una república unitaria y descentralizada que se divide administrativa y políticamente en 33 divisiones: 32 departamentos, los cuales son gobernados desde sus respectivas ciudades capitales, y un distrito capital, Bogotá. Los departamentos forman áreas culturales y económicas dentro de las regiones geográficas. En Colombia los recursos pasan de la nación al departamento y de este al municipio; a excepción de Bogotá, que como distrito capital, recibe directamente de la nación (a través de ley de regalías).

* **Censo Poblacional**

El censo de población y vivienda realizado, consiste en contar y caracterizar las personas residentes en Colombia, así como las viviendas y los hogares del territorio nacional. A través del censo, el país obtiene datos de primera mano sobre el número de habitantes, su distribución en el territorio y sus condiciones de vida.

Así, el censo no es solo una “fotografía” del país en un momento determinado, sino que permite generar información estadística que sirve para que diferentes organismos públicos y privados del país planeen y tomen decisiones de política pública, de desarrollo económico, bienestar social, empleo, vivienda, salud, migración, entre otros.

* **Censo Electoral**

La Ley 1475 de 2011, definió en su artículo 47 el censo electoral como “ el registro general de las cédulas de ciudadanía correspondientes a los ciudadanos colombianos, residentes en el país y en el exterior, habilitados por la Constitución y la ley para ejercer el derecho de sufragio y, por consiguiente, para participar en las elecciones y para concurrir a los mecanismos de participación ciudadana.
El censo electoral determina el número de electores que se requiere para la validez de los actos y votaciones. Es también el instrumento técnico, elaborado por la Registraduría Nacional del Estado Civil, que le permite a la Organización Electoral planear, organizar, ejecutar y controlar los certámenes electorales y los mecanismos de participación ciudadana”.

* **Participación Ciudadana**

A partir de la promulgación de la Constitución Política de 1991, Colombia se convirtió en un Estado Social de Derecho, fundada sobre la democracia participativa, en este sentido, la ciudadanía tiene el derecho y el deber de participar directamente en la concertación, consulta, control y vigilancia de las decisiones que los afecten y, en general, en la vida económica, política, administrativa y cultural del pueblo. Así mismo, los servidores públicos son llamados a promover y apoyar el derecho de la participación de los ciudadanos.

* **Abstención**

Abstención, en ciencia política, es el acto por el cual un potencial votante en unas elecciones decide no ejercer su derecho al voto, ya sea en unas elecciones generales o si, en un procedimiento parlamentario, el representante está presente en la votación pero no vota. En ambos casos, el sujeto se atiene al resultado del voto de los electores que sí votaron, al igual que ocurre con el voto en blanco.

Aunque la abstención no suele considerarse una de las opciones ante la toma colectiva de una decisión, una alta abstención se considera generalmente como beligerante o indiferente entre el cuerpo de votantes. Si la abstención se generaliza se pierde el bien público de la democracia

* **Votos Validos**

Son los votos depositados por cada uno de los candidatos inscritos más los votos en blanco. La suma de todos los votos depositados por los candidatos y votos en blanco arroja el total de votos válidos. No se incluyen dentro de los votos válidos los votos nulos ni las tarjetas no marcadas.

* **Votos en blanco**

De acuerdo con la sentencia C-490 de 2011 de la Corte Constitucional, que declaró la exequibilidad de la Ley 1475 (Reforma Política), el voto en blanco es “una expresión política de disentimiento, abstención o inconformidad, con efectos políticos” y agrega que “el voto en blanco constituye una valiosa expresión del disenso a través del cual se promueve la protección de la libertad del elector. Como consecuencia de este reconocimiento la Constitución le adscribe una incidencia decisiva en procesos electorales orientados a proveer cargos unipersonales y de corporaciones públicas de elección popular”.

* **Votos nulos**

Cuando la marcación efectuada por el elector no permite determinar de manera clara la intención del sufragante, es decir que no es posible identificar a favor de qué candidato votó.

* **Tarjetones no marcados**

La tarjeta no marcada es aquella en la cual a pesar de que el ciudadano la recibe, no la marca y la deposita en la urna sin haber seleccionado alguna opción. Por el contrario, la tarjeta no utilizada es aquella que al terminar la jornada queda en poder de los jurados y no fue entregada a los sufragantes. La tarjeta no marcada se contabiliza mientras que la no utilizada se destruye.





<img src="Jerarquia.png">


##  $\color{blue}{\text{Participación ciudadana}}$ </center>

En orden de jerarquia el primer aspecto a analizar es la demografía electoral presentada entorno a la participación ciudadana,analizar el crecimiento o decrecimiento de esta implica factores tanto sociales como inherentes.

Un primer factor extrinseco se derivada del crecimiento poblacional natural.En particular podemos verlo reflejado en nuestro intervalo temporal en análisis.

<img src="aumentopoblacion.png">

En específico,según la Registraduría Nacional del Estado Civil,las cifras evidencian que aunque el censo electoral creció de 1958 a 2010, en un promedio de 1.893.699 ciudadanos por cada elección presidencial,es decir en aproximadamente un 7.69% por periodo presidencial y en un 99.97% en los últimos 52 años, la participación política siempre ha fluctuado en el 56.52%.De los datos a continuación en el historico detallado se presenta un crecimiento congruente al población relativamente lineal en varias de las variables a analizar.

In [18]:
import pandas as pd
import openpyxl
wb = openpyxl.load_workbook('sufragiohistorico.xlsx')
ws = wb['Hoja1']
md = pd.DataFrame(ws.values)
md.columns = md.iloc[0]
md = md.drop(0)
md.head(27)

0           Año     Censo Votos válidos Votos en   Votos Tarjetas no  \
1          None      None          None   blanco   nulos    marcadas   
2          1958   5365191          None     8303    2165        None   
3          1962   5404765          None     5822  683436        None   
4          1966   6611352          None     9824    5529        None   
5          1970   7683785          None    36892    5426        None   
6          1974   8925330          None     6722    5869        None   
7          1978  12580851          None     9923    7871        None   
8          1982  13734093       6815660     8996    9594        None   
9          1986  15611274          None    42205    8119        None   
10         1990  14237110       6002274    77727   45302        None   
11         1994  17146597       5791332    65116   29999        None   
12  (1ª vuelta)      None          None     None    None        None   
13         1994  17146597       7382653    72536   45089       43446   
14  (2ª vuelta)      None          None     None    None        None   
15         1998  20857801      10630623   122431   79396       54384   
16  (1ª vuelta)      None          None     None    None        None   
17         1998  20857801      12146929   373659  108794       48966   
18  (2ª vuelta)      None          None     None    None        None   
19         2002  24208311      11051645   196116  149123        None   
20         2006  26731700      11864410   226297  132332       44995   
21         2010  29983279      14572593   223977  170874       37553   
22  (1ª vuelta)      None          None     None    None        None   
23         2010  29983279      13061192   444274  198003       37729   
24  (2ª vuelta)      None          None     None    None        None   

0   Votación Participación Abstención  
1      total          None       None  
2    3108567         57.71      42.29  
3    2634840         48.75      51.25  
4    2649258         40.07      59.93  
5    4036458         52.53      47.47  
6    5218855         58.47      41.53  
7    5075719         40.34      59.66  
8    6834250         49.76      50.24  
9    7228676          46.3       53.7  
10   6047576         42.48      57.52  
11   5821331         33.95      66.05  
12      None          None       None  
13   7427742         43.32      56.68  
14      None          None       None  
15  10753465         51.56      48.44  
16      None          None       None  
17  12310107         40.99      59.01  
18      None          None       None  
19   1149734         46.47      53.53  
20  12041737         45.05      54.95  
21  14781020         49.29      50.71  
22      None          None       None  
23  13296924         44.34      55.66  
24      None          None       None

Ahora bien,todas estas cifras aunque presentan un cercanía cada una de ellas fueron producidas en contextos diferentes tanto en las formas de sufragio como en las politicas internas,los factores sociales que directa o indirectamente anclan el voto popular como por ejemplo problemas sociales,varios sonados fuertemente en las ultimas decadas,de allí la necesidad de relacionar las cifras con el contexto temporal en el que se encuentran,nuestro periodo a analizar naturalmente es el inmediatamente mas cercano en donde se evidencia el siguiente incremento de la participación y decrecimiento del abstencionismo.

<img src="aumentosufragio.png">

Veamos especificamente los comportamientos de cada uno de estos dos periodos en sus respectivas dos contiendas y en los distintos departamentos del pais

$2014$

<img src="20141.png">

<img src="20142.png">

$2018$

<img src="20181.png">

<img src="20182.png">

En los anteriores gráficos de barras se refleja la consistencia entre vueltas electorales por periodo y al igual se evidencia la concentración de poder de voto en algunos de los departamentos del pais.Para 2018 naturalmente se conserva la uniformidad en el baloteje sin embargo se refleja un importante alza tanto en el censo como en la participación política

Especificamente la distribución mantiene una tradicional consistencia la cual se reparte en los territorios de la siguiente forma

<img src="concentracionp.png">

Relacionando los fenomenos de concentración de poder y alza de participación podemos evidenciar la importancia de los centros más poblados objetivos siempre de candidatos y partidos.

<img src="pordepartamento.png">


##  $\color{blue}{\text{Transiciones y transferencias electorales asociadas a Partidos Políticos}}$ </center>

$\color{blue}{\text{2014}}$

$Primera\;Vuelta$

In [19]:
votos=['Votos Oscar Iván Zuluaga',
 'Votos Juan Manuel Santos',
 'Votos Martha Lucía Ramírez','Votos Clara López Obregón','Votos Enrique Peñalosa']
Datos_10_2014[votos].sum()

Votos Oscar Iván Zuluaga      3769005
Votos Juan Manuel Santos      3310794
Votos Martha Lucía Ramírez    1997980
Votos Clara López Obregón     1958518
Votos Enrique Peñalosa        1064758
dtype: int64

$Segunda\;Vuelta$

In [20]:
votos=['Juan Manuel Juan Manuel Santos',
 'Oscar Iván Oscar Iván Zuluaga']
Datos_20_2014[votos].sum()

Juan Manuel Juan Manuel Santos    7773117
Oscar Iván Oscar Iván Zuluaga     6840973
dtype: int64

|![alt](mapaelectoral20141.png), | ![alt](mapaelectoral20142.png) |
|-|-|
|![alt](labelmapa14.png), |

$\color{blue}{\text{2018}}$

$Primera\;Vuelta$

In [21]:
votos=['Iván Duque','Gustavo Petro','Sergio Fajardo','Germán Vargas Lleras','Humberto De La Calle','Jorge Antonio Trujillo ','Viviane Morales']
Datos_10_2018[votos].sum()

Iván Duque                 7617114.0
Gustavo Petro              4854831.0
Sergio Fajardo             4602916.0
Germán Vargas Lleras       1412392.0
Humberto De La Calle        396151.0
Jorge Antonio Trujillo       65767.0
Viviane Morales              36138.0
dtype: float64

In [22]:
votos=['Iván Duque','Gustavo Petro']
Datos_20_2018[votos].sum()

Iván Duque       10398689
Gustavo Petro     8039504
dtype: int64

|![alt](mapaelectoral20181.png), | ![alt](mapaelectoral20182.png) |
|-|-|
|![alt](labelmapa18.png), |

TEXTO DOMINANCIA BILATERAL

## $\color{darkblue}{\text{Centro Democrático}}$

$2014-1$

In [23]:
Depto=Datos_10_2014.groupby(['Depto'])['Votos Oscar Iván Zuluaga'].sum()
Depto.sort_values(ascending=False).head()

Depto
Antioquia          665160
Bogotá, D.C.       542459
Cundinamarca       255598
Valle del Cauca    216620
Santander          186420
Name: Votos Oscar Iván Zuluaga, dtype: int64

$2018-1$

In [24]:
Depto=Datos_10_2018.groupby(['Depto'])['Iván Duque'].sum()
Depto.sort_values(ascending=False).head()


Depto
Antioquia          1375965.0
Bogotá D.C.         989744.0
Valle del Cauca     527622.0
Cundinamarca        479936.0
Santander           442219.0
Name: Iván Duque, dtype: float64

## $\color{orange}{\text{Antioquia}}$

2014

In [25]:
votos=['Votos válidos','Votos Oscar Iván Zuluaga',
 'Votos Juan Manuel Santos',
 'Votos Martha Lucía Ramírez','Votos Clara López Obregón','Votos Enrique Peñalosa']
Antioquia=Datos_10_2014[Datos_10_2014['Depto']=='Antioquia']
ANT=Antioquia[votos].sum()
df=pd.DataFrame(ANT)
porc=df.assign(Porcentaje=ANT/df.iloc[0,0]*100)
porc

0  Porcentaje
Votos válidos               1766390  100.000000
Votos Oscar Iván Zuluaga     665160   37.656463
Votos Juan Manuel Santos     286742   16.233221
Votos Martha Lucía Ramírez   334312   18.926285
Votos Clara López Obregón    248628   14.075487
Votos Enrique Peñalosa        98144    5.556191

In [26]:
votos=['Votos Válidos','Oscar Iván Oscar Iván Zuluaga','Juan Manuel Juan Manuel Santos']
Antioquia=Datos_20_2014[Datos_20_2014['Departamento']=='Antioquia']
ANT=Antioquia[votos].sum()
df=pd.DataFrame(ANT)
porc=df.assign(Porcentaje=ANT/df.iloc[0,0]*100)
porc

0  Porcentaje
Votos Válidos                   1967570  100.000000
Oscar Iván Oscar Iván Zuluaga   1137735   57.824372
Juan Manuel Juan Manuel Santos   704164   35.788511

2018

In [27]:
votos=['Votos Válidos','Iván Duque','Gustavo Petro','Sergio Fajardo','Germán Vargas Lleras','Humberto De La Calle','Jorge Antonio Trujillo ','Viviane Morales']
Antioquia=Datos_10_2018[Datos_10_2018['Depto']=='Antioquia']
ANT=Antioquia[votos].sum()
df=pd.DataFrame(ANT)
porc=df.assign(Porcentaje=ANT/df.iloc[0,0]*100)
porc

0  Porcentaje
Votos Válidos            2575391.0  100.000000
Iván Duque               1375965.0   53.427421
Gustavo Petro             237206.0    9.210485
Sergio Fajardo            732993.0   28.461426
Germán Vargas Lleras       99182.0    3.851143
Humberto De La Calle       59136.0    2.296195
Jorge Antonio Trujillo      6985.0    0.271221
Viviane Morales             4398.0    0.170770

In [28]:
votos=['Votos Válidos','Iván Duque','Gustavo Petro']
Antioquia=Datos_20_2018[Datos_20_2018['Depto']=='Antioquia']
ANT=Antioquia[votos].sum()
df=pd.DataFrame(ANT)
porc=df.assign(Porcentaje=ANT/df.iloc[0,0]*100)
porc

0  Porcentaje
Votos Válidos  2547294  100.000000
Iván Duque     1850996   72.665189
Gustavo Petro   556685   21.853975

<img src="Centrodemocratico.png">

## $\color{orange}{\text{Cundinamarca}}$

2014

In [29]:
votos=['Votos válidos','Votos Oscar Iván Zuluaga',
 'Votos Juan Manuel Santos',
 'Votos Martha Lucía Ramírez','Votos Clara López Obregón','Votos Enrique Peñalosa']
Cundinamarca=Datos_10_2014[Datos_10_2014['Depto']=='Cundinamarca']
CUN=Cundinamarca[votos].sum()
df=pd.DataFrame(CUN)
porc=df.assign(Porcentaje=CUN/df.iloc[0,0]*100)
porc

0  Porcentaje
Votos válidos               808109  100.000000
Votos Oscar Iván Zuluaga    255598   31.629149
Votos Juan Manuel Santos    144346   17.862194
Votos Martha Lucía Ramírez  186690   23.102082
Votos Clara López Obregón   116562   14.424044
Votos Enrique Peñalosa       56156    6.949063

In [30]:
votos=['Votos Válidos','Oscar Iván Oscar Iván Zuluaga','Juan Manuel Juan Manuel Santos']
Cundinamarca=Datos_20_2014[Datos_20_2014['Departamento']=='Antioquia']
CUN=Cundinamarca[votos].sum()
df=pd.DataFrame(CUN)
porc=df.assign(Porcentaje=CUN/df.iloc[0,0]*100)
porc

0  Porcentaje
Votos Válidos                   1967570  100.000000
Oscar Iván Oscar Iván Zuluaga   1137735   57.824372
Juan Manuel Juan Manuel Santos   704164   35.788511

2018

In [31]:
votos=['Votos Válidos','Iván Duque','Gustavo Petro','Sergio Fajardo','Germán Vargas Lleras','Humberto De La Calle','Jorge Antonio Trujillo ','Viviane Morales']
Cundinamarca=Datos_10_2018[Datos_10_2018['Depto']=='Cundinamarca']
CUN=Cundinamarca[votos].sum()
df=pd.DataFrame(CUN)
porc=df.assign(Porcentaje=CUN/df.iloc[0,0]*100)
porc

0  Porcentaje
Votos Válidos            1185853.0  100.000000
Iván Duque                479936.0   40.471795
Gustavo Petro             236175.0   19.916044
Sergio Fajardo            329857.0   27.816011
Germán Vargas Lleras       85943.0    7.247357
Humberto De La Calle       20545.0    1.732508
Jorge Antonio Trujillo      4185.0    0.352911
Viviane Morales             2611.0    0.220179

In [32]:
votos=['Votos Válidos','Iván Duque','Gustavo Petro']
Cundinamarca=Datos_20_2018[Datos_20_2018['Depto']=='Cundinamarca']
CUN=Cundinamarca[votos].sum()
df=pd.DataFrame(CUN)
porc=df.assign(Porcentaje=CUN/df.iloc[0,0]*100)
porc

0  Porcentaje
Votos Válidos  1166553  100.000000
Iván Duque      661521   56.707325
Gustavo Petro   450382   38.607933

## $\color{purple}{\text{Colombia Humana}}$

In [33]:
Depto=Datos_10_2018.groupby(['Depto'])['Gustavo Petro'].sum()
Depto.sort_values(ascending=False).head()


Depto
Bogotá D.C.        1099955
Valle del Cauca     485867
Atlántico           331979
Córdoba             250437
Bolívar             242471
Name: Gustavo Petro, dtype: int64

## $\color{orange}{\text{Bogotá}}$

2018-I

In [34]:
votos=['Votos Válidos','Sergio Fajardo','Gustavo Petro','Iván Duque','Germán Vargas Lleras','Humberto De La Calle','Jorge Antonio Trujillo ','Viviane Morales']
Bogota=Datos_10_2018[Datos_10_2018['Depto']=='Bogotá D.C.']
BOG=Bogota[votos].sum()
df=pd.DataFrame(BOG)
porc=df.assign(Porcentaje=BOG/df.iloc[0,0]*100)
porc

0  Porcentaje
Votos Válidos            3676217.0  100.000000
Sergio Fajardo           1244477.0   33.852109
Gustavo Petro            1099955.0   29.920840
Iván Duque                989744.0   26.922894
Germán Vargas Lleras      172684.0    4.697329
Humberto De La Calle       93401.0    2.540682
Jorge Antonio Trujillo     10696.0    0.290951
Viviane Morales             5492.0    0.149393

2018-II

In [35]:
votos=['Votos Válidos','Gustavo Petro','Iván Duque']
Bogota=Datos_20_2018[Datos_20_2018['Depto']=='Bogotá D.C.']
BOG=Bogota[votos].sum()
df=pd.DataFrame(BOG)
porc=df.assign(Porcentaje=BOG/df.iloc[0,0]*100)
porc

0  Porcentaje
Votos Válidos  3538391  100.000000
Gustavo Petro  1889050   53.387260
Iván Duque     1449092   40.953416

<img src="ColombiaHumana.png">

## $\color{orange}{\text{Valle del Cauca}}$

2018-I

In [36]:
votos=['Votos Válidos','Iván Duque','Sergio Fajardo','Gustavo Petro','Germán Vargas Lleras','Humberto De La Calle','Jorge Antonio Trujillo ','Viviane Morales']
Valle=Datos_10_2018[Datos_10_2018['Depto']=='Valle del Cauca']
VAL=Valle[votos].sum()
df=pd.DataFrame(VAL)
porc=df.assign(Porcentaje=VAL/df.iloc[0,0]*100)
porc

0  Porcentaje
Votos Válidos            1747769.0  100.000000
Iván Duque                527622.0   30.188314
Sergio Fajardo            502367.0   28.743329
Gustavo Petro             485867.0   27.799269
Germán Vargas Lleras      137373.0    7.859906
Humberto De La Calle       40858.0    2.337723
Jorge Antonio Trujillo      6852.0    0.392043
Viviane Morales             2971.0    0.169988

2018-II

In [37]:
votos=['Votos Válidos','Gustavo Petro','Iván Duque']
Valle=Datos_20_2018[Datos_20_2018['Depto']=='Valle del Cauca']
VAL=Valle[votos].sum()
df=pd.DataFrame(VAL)
porc=df.assign(Porcentaje=VAL/df.iloc[0,0]*100)
porc

0  Porcentaje
Votos Válidos  1713969  100.000000
Gustavo Petro   888023   51.810914
Iván Duque      747733   43.625818

<img src="valledelcauca.png">

## <center> $\color{teal}{\text{Especificación del modelo de ML}}$ </center>

$\color{teal}{\text{Motivación del proyecto de ML}}$

Como lo vimos en los datos del proyecto ‘Análisis del panorama político para las elecciones presidenciales de 2022 ’estos no solo ofrecen la cantidad de votos obtenidos por un candidato, sino que además generan todo un entorno y caracterización demográfica respecto al entorno político-social y preferencias ciudadanas, el cual resulta de gran importancia poder observar y comprender.
Así mismo, dentro del desarrollo del proyecto se planteó el contraste cuantitativo como medida de desempeño y éxito de las cifras obtenidas de las distintas variables a analizar frente a los sondeos actuales realizados por las firmas encuestadoras donde si bien se posee un alto nivel de congruencia respecto a las proyecciones y los resultados en las urnas también se presenta un margen de error considerable pues aunque las firmas encuestadoras intentan tomar un grupo muestral característico este aún puede verse aún reducido en comparación con el tamaño real de la población .
Hoy en día el Machine Learning es fuertemente aplicado en el entorno social, desde la publicidad mostrada a cada usuario hasta el mismo algoritmo que los clasifica entre sí y por lo tanto el ámbito político no se queda atrás pues hoy en día las redes sociales son el primer lugar para divulgar las posturas respecto a algún tema y en particular políticas.
La aplicación del presente proyecto al Machine Learning que se pretende abordar es como las redes sociales pueden establecer métricas más exactas de las posturas políticas de los individuos tomando no solo un grupo mas cuantioso sino además con mayor activismo y presencia en divulgación e influencia clave en la sociedad de hoy en día.


<img src="MLUSA.jpg">

$\color{teal}{\text{Definición del problema}}$

$\color{teal}{\text{Salida}}$

La buena ejecución del problema de ML debería proporcionar si una persona es afín o no a un candidato o hay una posible intención de voto de los distintos ciudadanos en los diferentes territorios y departamentos a un candidato proporcionando una polaridad positiva negativa o neutra, esto de forma análoga a lo obtenido en la analítica de datos, pero esta vez mediante el análisis de las opiniones y comentarios publicados en tiempo real en las distintas redes sociales de gran flujo de texto y fácil accesibilidad como por ejemplo Twitter.

$\color{teal}{\text{Entrada}}$

Para la recopilación de la entrada de datos a trabajar en el proceso de analítica de datos en este caso puntual los tweets, se deben tener en cuenta distintos factores de selección y clasificación, esto con el fin de obtener una mayor calidad en la información en materia de magnitud y contenido, algunos de los más relevantes para tener en cuenta podrían ser:

-	**Selección reciente:**
    
    Si bien una parte de la población es fiel a sus posturas unas las cuestiones que mas se analiza en una contienda electoral es el ‘sentimiento popular’ y la variación de este por lo cual un escaneo a la red de a lo mas 3-4 meses previos a las elecciones contienen información más sólida.

- **Mayor cantidad de trafico social**

    Existen horas del día donde el trafico de redes sociales es mucho mayor comparado con otras franjas horarias y esto en política fuertemente conectado cuando los medios de comunican masiva se pronuncian respecto a algún tema que de cierta forma directa o indirecta atañe a algún candidato o partido político y lo anterior cruzado cuando las personas tienen más presencia en sus redes como lo son las horas sociales y no laborales.

- **Anclaje a eventos políticos**

    Este se deriva del factor inmediatamente anterior y es que cuando algún personaje político se pronuncia o actúa de determinada forma respecto a algún tema esto genera una ola de comentarios y reacciones de con alto flujo y expansión en la red.

- **Geolocalización**

    Resulta mas que necesario seccionar los datos por territorios pues de esto depende el dominio de un candidato en un territorio pues esto se logra en la obtención de un sufragante y la expansión en sus círculos sociales más próximos como los son familiares y amigos ,aunque si bien cada voto cuenta por el análisis de datos visto en el proyecto la concentración del poder de voto se encuentra en las principales capitales; De igual forma el histórico de afinidad de partidos políticos a  departamentos 

Una vez recopilados los tweets producidos en determinado tiempo debe ser posible filtrarlos para solo seleccionar los que son afines a nuestro estudio político y esto se logra mediante el previo entrenamiento del modelo el cual claramente debe ser supervisado pues la conexión léxica con las distintas expresiones y la biblioteca de posibles nexos para deducir un sentimiento de intención de voto es bastante amplia.

$\color{teal}{\text{DataSet}}$

$\color{black}{\text{Número de registros de entrenamiento}}$

Establecer un numero de entrenamientos es algo nada rudimentario sin embargo es claro que este debe ser un considerable diccionario de palabras anclado a los temas y sujetos de las próximas contiendas, algunos de estos a considerar deben ser:

$\color{black}{\text{Para la obtención:}}$

<img src="MLDATASET1.png">

$\color{black}{\text{Para la filtración:}}$

<img src="MLDATASET2.png">

$\color{teal}{\text{Datos no balanceados y sesgos}}$

Un posible sesgo sistemático puede provenir de la cantidad de interpretaciones que pueden tener hoy en día las expresiones incluso por más cortas que estas sean, en el trabajo de toma y filtración se deben tener varias capas de procesamiento que consideren o descarten situaciones como:

•	Monosílabos sin conexión, cadenas de caracteres ilegibles

•	Tweets que no presentan conexión con algún tema político

•	Imágenes 

•	Links, sitios de redirección, tweets automatizados de página clasificadas como Spam

•	Expresiones coloquiales o determinadas formas de referirse a una cosa, tema o sujeto de una forma no tan literal

•	Expresiones en tono irónico o subjetivo donde no se refleje con inmediata claridad una verdadera favorabilidad o desfavorabilidad 


Si bien estos filtros pueden descartar el posible ruido de la cantidad de datos también están las formas directas de anclar los tweets de importancia directa como los son nombrar de forma explícita al candidato, así como las tendencias y hashtags

Otros posibles sesgos mas ligados al sujeto pueden ser:

•	Cuentas de menores de edad

•	Cuentas ligadas a empresas u organizaciones caracterizadas por la difusión de contenido y que no pertenecen a ningún sujeto en particular


$\color{teal}{\text{Frecuencia}}$

Como lo dicho anteriormente es necesario fijar un periodo cercano a las elecciones, dentro de este periodo se podría considerar un escaneo semanal en una franja horaria pues esto nada más genera una gran cantidad de datos, la periodicidad  ayudaría aportando no solo a variabilidad de población y cantidad de sujetos sino además tomando en cuenta los eventos o situaciones claves que van ocurriendo en momentos claves como los son los cierres de campañas que derivan eventos masivos y posturas de diferentes colectivos y gremios.

$\color{teal}{\text{Predicción}}$

Tras una selección y filtración de calidad en los datos recopilados en la red el siguiente paso clave es la clasificación, el entendimiento de la información permitirá categorizar complementariamente en positivo o negativo, favorable o desfavorables, de donde categorizando cada opinión y tomando un buen tamaño muestral se podría recopilar y dar un porcentaje de que dado un total de población que fracción de esta votaría por determinado candidato. Como veremos más adelante en la literatura relacionada con el problema se presenta fuertemente la conexión entre el clasificador Naive-Bayes junto con la estimación y valoración que dan la herramienta crucial a los algoritmos para finalmente arrojar un dato cuantitativo que pueda ser traducido a los distintos actores interesados en el problema

In [38]:
import twint

In [39]:
URL='Prueba1.xlsx'
HOJA='Prueba'
Tweets=pd.read_excel(URL,sheet_name=HOJA)

In [40]:
Tweets

id      conversation_id  \
0    1433749039011529984  1433575210037509888   
1    1433745090514660096  1433543355934320128   
2    1433678054715529984  1432796927901449984   
3    1433659794867589888  1433590174747930112   
4    1433629505483120128  1432312227386850048   
..                   ...                  ...   
224  1432997837688780032  1432717887848979968   
225  1432952184832920064  1432847808411930112   
226  1432951051242569984  1432881943834670080   
227  1432941894082899968  1432707120982599936   
228  1432935341011389952  1432883830453639936   

                                            created_at       date      time  \
0    2021-09-03 06:09:43 Hora est. Pacífico, Sudam... 2021-09-03  06:09:43   
1    2021-09-03 05:54:01 Hora est. Pacífico, Sudam... 2021-09-03  05:54:01   
2    2021-09-03 01:27:39 Hora est. Pacífico, Sudam... 2021-09-03  01:27:39   
3    2021-09-03 00:15:05 Hora est. Pacífico, Sudam... 2021-09-03  00:15:05   
4    2021-09-02 22:14:44 Hora est. Pacífico, Sudam... 2021-09-02  22:14:44   
..                                                 ...        ...       ...   
224  2021-09-01 04:24:42 Hora est. Pacífico, Sudam... 2021-09-01  04:24:42   
225  2021-09-01 01:23:18 Hora est. Pacífico, Sudam... 2021-09-01  01:23:18   
226  2021-09-01 01:18:48 Hora est. Pacífico, Sudam... 2021-09-01  01:18:48   
227  2021-09-01 00:42:24 Hora est. Pacífico, Sudam... 2021-09-01  00:42:24   
228  2021-09-01 00:16:22 Hora est. Pacífico, Sudam... 2021-09-01  00:16:22   

     timezone              user_id         username                     name  \
0        -500           1395739237       shello1975                 Ey barro   
1        -500  1254547922471000064  mercede99950857       Mercedes Hernandez   
2        -500  1242245953006030080  jairoca87518405            Jairo Camacho   
3        -500            118488815          faventi                  Faventi   
4        -500  1400535100425649920   almadecolores1  Alma de Colores 🦋✨   
..        ...                  ...              ...                      ...   
224      -500            790482266    pablodupablos               |-/ pablø   
225      -500  1068928388675570048   andantesinmoto                     🤓   
226      -500  1392648880056110080  colombi55834121  Una Colombia de Derecha   
227      -500            555370282     imgranadosr1                mgranados   
228      -500            134903584     malibucaribe       yusti de los Reyes   

     place  ... geo source user_rt_id user_rt retweet_id  \
0      NaN  ... NaN    NaN        NaN     NaN        NaN   
1      NaN  ... NaN    NaN        NaN     NaN        NaN   
2      NaN  ... NaN    NaN        NaN     NaN        NaN   
3      NaN  ... NaN    NaN        NaN     NaN        NaN   
4      NaN  ... NaN    NaN        NaN     NaN        NaN   
..     ...  ...  ..    ...        ...     ...        ...   
224    NaN  ... NaN    NaN        NaN     NaN        NaN   
225    NaN  ... NaN    NaN        NaN     NaN        NaN   
226    NaN  ... NaN    NaN        NaN     NaN        NaN   
227    NaN  ... NaN    NaN        NaN     NaN        NaN   
228    NaN  ... NaN    NaN        NaN     NaN        NaN   

                                              reply_to  retweet_date  \
0    [{'screen_name': 'sergio_fajardo', 'name': 'Se...           NaN   
1    [{'screen_name': 'agaviriau', 'name': 'Alejand...           NaN   
2    [{'screen_name': 'agaviriau', 'name': 'Alejand...           NaN   
3    [{'screen_name': 'alinpinpri', 'name': 'El gar...           NaN   
4    [{'screen_name': 'CharlyDoval', 'name': 'Rinho...           NaN   
..                                                 ...           ...   
224  [{'screen_name': 'juankyospina', 'name': 'Juan...           NaN   
225  [{'screen_name': 'ArgiroCasta58', 'name': 'Arg...           NaN   
226  [{'screen_name': 'marcelamvz', 'name': 'marce ...           NaN   
227  [{'screen_name': 'santiagoecheve', 'name': 'Sa...           NaN   
228  [{'sc

In [41]:
df=pd.DataFrame(Tweets['tweet'])

In [42]:
df

tweet
0    @sergio_fajardo @agaviriau @CoaliEsperanza Ni ...
1    @agaviriau Gran esperanza para colombia! OjalÃ...
2    @agaviriau Usted hace parte del continuismo,no...
3    @alinpinpri @JuanAngelTin @NeiderCamargo5 @Dav...
4    @CharlyDoval @franja_uno @AndresCamiloHR @ciel...
..                                                 ...
224  @juankyospina @chococongas @RevistaSemana Y @p...
225  @ArgiroCasta58 Me choca ver a la gente posando...
226  @marcelamvz Yo te digo. Yo voto en 1era vuelta...
227  @santiagoecheve @todoscolombia @petrogustavo @...
228  @petrogustavo Soy liberal pero mi Voto es por ...

[229 rows x 1 columns]

In [43]:
import re
def clean_tweets(text):
  text = re.sub("RT @[\w]*:","",text)
  #text = re.sub("@[\w]*","",text)
  text = re.sub("https?://[A-Za-z0-9./]*","",text)
  text = re.sub("\n","",text)
  return text
df['tweet'] = df['tweet'].apply(lambda x: clean_tweets(x))
df.head(20)

tweet
0   @sergio_fajardo @agaviriau @CoaliEsperanza Ni ...
1   @agaviriau Gran esperanza para colombia! OjalÃ...
2   @agaviriau Usted hace parte del continuismo,no...
3   @alinpinpri @JuanAngelTin @NeiderCamargo5 @Dav...
4   @CharlyDoval @franja_uno @AndresCamiloHR @ciel...
5   @DiegoTrujilloD @agaviriau Claaaaaro, ya sabem...
6   @MaoS_777 @sergio_fajardo @agaviriau @CoaliEsp...
7   @sergio_fajardo @agaviriau @CoaliEsperanza Si ...
8   @DiegoTrujilloD @agaviriau Este es otro que vo...
9   @agaviriau Solo puedo decirle algo, de ganar, ...
10  @plagiosos @beticovergara @agaviriau Entonces ...
11  @agaviriau Excelente ser humano , mi voto por ...
12       @DanielSamperO 10. Mi voto es por @agaviriau
13  @DiegoTrujilloD @agaviriau Gaviria es la nueva...
14  @DiegoTrujilloD @agaviriau Petardo, otra vez v...
15  @DiegoTrujilloD @agaviriau Alejandro me habla ...
16  @sergio_fajardo @CoaliEsperanza @agaviriau Es ...
17  @JuanitaGoe @agaviriau HE VOTADO en ocasiones ...
18  @OsComSam @GNavasTalero @JulianEnriqueGo @agav...
19  @mariavi03638326 @antamanece @FicoGutierrez @a...

In [44]:
from textblob import TextBlob

In [45]:
TextBlob ('Odio a Petro').translate(from_lang='es',to='en').sentiment

Sentiment(polarity=-0.8, subjectivity=0.9)

In [76]:
TextBlob ('Mejor opcion petro').translate(from_lang='es',to='en').sentiment

Sentiment(polarity=1.0, subjectivity=0.3)

In [77]:
TextBlob ('Peor opcion petro').translate(from_lang='es',to='en').sentiment

Sentiment(polarity=-1.0, subjectivity=1.0)

In [47]:
import time
def get_polarity(text):
  analysis = TextBlob(text)
  if text != '':
    if analysis.detect_language() == 'es':
      result = analysis.translate(from_lang = 'es', to = 'en').sentiment.polarity
      time.sleep(5)
      return result

In [72]:
A = df[1:30]
#df['polarity'] = df['tweet'].apply(get_polarity)

In [73]:
A

tweet
1   @agaviriau Gran esperanza para colombia! OjalÃ...
2   @agaviriau Usted hace parte del continuismo,no...
3   @alinpinpri @JuanAngelTin @NeiderCamargo5 @Dav...
4   @CharlyDoval @franja_uno @AndresCamiloHR @ciel...
5   @DiegoTrujilloD @agaviriau Claaaaaro, ya sabem...
6   @MaoS_777 @sergio_fajardo @agaviriau @CoaliEsp...
7   @sergio_fajardo @agaviriau @CoaliEsperanza Si ...
8   @DiegoTrujilloD @agaviriau Este es otro que vo...
9   @agaviriau Solo puedo decirle algo, de ganar, ...
10  @plagiosos @beticovergara @agaviriau Entonces ...
11  @agaviriau Excelente ser humano , mi voto por ...
12       @DanielSamperO 10. Mi voto es por @agaviriau
13  @DiegoTrujilloD @agaviriau Gaviria es la nueva...
14  @DiegoTrujilloD @agaviriau Petardo, otra vez v...
15  @DiegoTrujilloD @agaviriau Alejandro me habla ...
16  @sergio_fajardo @CoaliEsperanza @agaviriau Es ...
17  @JuanitaGoe @agaviriau HE VOTADO en ocasiones ...
18  @OsComSam @GNavasTalero @JulianEnriqueGo @agav...
19  @mariavi03638326 @antamanece @FicoGutierrez @a...
20  @sergio_fajardo @CoaliEsperanza @agaviriau No ...
21  @agaviriau Tu candidatura trae vientos de espe...
22  @PelosNecios23 @gastonvelandia @JERobledo @Coa...
23  Señor @agaviriau le preguntaré hasta el últ...
24  @FelipeB83 @EdicsonMuoz @sergio_fajardo @Coali...
25  @marlonacr @jrrangelp @angelamrobledo @wilmarv...
26  @alberto_cabo @agaviriau Lo que da escalofrío...
27  @sgcs2324 @CarlosAmayaR @petrogustavo La verda...
28  @sergio_fajardo @CoaliEsperanza @agaviriau Agr...
29  @JavierAdaime @Jokeraton @agaviriau @alcarrasq...

In [74]:
A['tweet'].apply(get_polarity)

1     1.000000
2     0.017857
3     0.000000
4     0.000000
5     0.000000
6     0.295312
7    -0.002381
8     0.250000
9     0.400000
10    0.000000
11    0.500000
12    0.000000
13   -0.231818
14   -0.250000
15    0.216667
16    0.133333
17   -0.200000
18   -0.100000
19    0.000000
20    0.000000
21    0.066667
22    0.000000
23    0.100000
24    0.000000
25   -0.333333
26   -0.650000
27   -0.127500
28    0.250000
29    0.000000
Name: tweet, dtype: float64